app/assets/javascripts/discourse/app/components/post/menu.gjs

```diff
@@ -0,0 +1,603 @@
+      component: AdminPostMenu,
+      modalForMobile: true,
+      autofocus: true,
+      data: {
+        post: this.args.post,
+        changeNotice: this.args.changeNotice,
+        changePostOwner: this.args.changePostOwner,
+        grantBadge: this.args.grantBadge,
+        lockPost: this.args.lockPost,
+        permanentlyDeletePost: this.args.permanentlyDeletePost,
+        rebakePost: this.args.rebakePost,
+        showPagePublish: this.args.showPagePublish,
+        togglePostType: this.args.togglePostType,
+        toggleWiki: this.args.toggleWiki,
+        unhidePost: this.args.unhidePost,
+        unlockPost: this.args.unlockPost,
+      },
+    });
+  }
+
+  @action
+  refreshReaders() {
+    if (this.readers.length) {
+      return this.#fetchWhoRead();
+    }
+  }
+
+  @action
+  showDeleteTopicModal() {
+    this.modal.show(DeleteTopicDisallowedModal);
+  }
+
+  @action
+  async showMoreActions() {
+    this.collapsed = false;
+
+    const fetchData = [
+      !this.isWhoLikedVisible && this.#fetchWhoLiked(),
+      !this.isWhoReadVisible &&
+        this.args.showReadIndicator &&
+        this.#fetchWhoRead(),
+    ].filter(Boolean);
+
+    await Promise.all(fetchData);
+  }
+
+  @action
+  toggleWhoLiked() {
+    if (this.isWhoLikedVisible) {
+      this.isWhoLikedVisible = false;
+      return;
+    }
+
+    this.#fetchWhoLiked();
+  }
+
+  @action
+  toggleWhoRead() {
+    if (this.isWhoReadVisible) {
+      this.isWhoReadVisible = false;
+      return;
+    }
+
+    this.#fetchWhoRead();
+  }
+
+  get #hiddenItems() {
+    const setting = this.siteSettings.post_menu_hidden_items;
+
+    if (isEmpty(setting)) {
+      return [];
+    }
+
+    return setting
+      .split("|")
+      .filter(
+        (itemKey) =>
+          !this.args.post.bookmarked ||
+          itemKey !== POST_MENU_BOOKMARK_BUTTON_KEY
+      );
+  }
+
+  async #fetchWhoLiked() {
+    const users = await this.store.find("post-action-user", {
+      id: this.args.post.id,
+      post_action_type_id: LIKE_ACTION,
+    });
+
+    this.likedUsers = users.map(smallUserAttributes);
+    this.totalLikedUsers = users.totalRows;
+    this.isWhoLikedVisible = true;
+  }
+
+  async #fetchWhoRead() {
+    const users = await this.store.find("post-reader", {
+      id: this.args.post.id,
+    });
+
+    this.readers = users.map(smallUserAttributes);
+    this.totalReaders = users.totalRows;
+    this.isWhoReadVisible = true;
+  }
+
+  <template>
+    {{! The section tag can't be include while we're still using the widget shim }}
+    {{! <section class="post-menu-area clearfix"> }}
+    <PluginOutlet
+      @name="post-menu"
+      @outletArgs={{hash post=@post state=this.state}}
+    >
+      <nav
+        class={{concatClass
+          "post-controls"
+          "glimmer-post-menu"
+          (if
+            (and
+              (this.showMoreButton.shouldRender
+                (hash post=this.post state=this.state)
+              )
```

Comment:
could you extract it to a getter? calling `shouldRender` like a helper means it won't have the correct `this` context, so you won't be able to e.g. call other static methods on a given class

Reply:
This is not an issue, `this.showMoreButton` in this case is a `PostMenuButtonConfig` instance.

If we were indeed calling the static method we'd also have to pass the other parameters: `(args, context, owner)`. :-)

The static method `shouldRender` is called internally by the button config `shouldRender` method.

https://github.com/discourse/discourse/blob/6e7f27648fa58b1c98ad08dddadd701fdf75c433/app/assets/javascripts/discourse/app/components/post/menu/button-config.gjs#L64-L71